test/framework/src/main/java/org/elasticsearch/transport/MockTransportClient.java

```diff
@@ -37,19 +40,34 @@ public MockTransportClient(Settings settings, Class<? extends Plugin>... plugins
     }
 
     public MockTransportClient(Settings settings, Collection<Class<? extends Plugin>> plugins) {
-        this(settings, addMockTransportIfMissing(plugins), null);
+        this(settings, plugins, null);
     }
 
     public MockTransportClient(Settings settings, Collection<Class<? extends Plugin>> plugins, HostFailureListener listener) {
-        super(settings, DEFAULT_SETTINGS, addMockTransportIfMissing(plugins), listener);
+        super(settings, DEFAULT_SETTINGS, addMockTransportIfMissing(settings, plugins), listener);
     }
 
-    private static Collection<Class<? extends Plugin>> addMockTransportIfMissing(Collection<Class<? extends Plugin>> plugins) {
-        if (plugins.contains(MockTcpTransportPlugin.class)) {
-            return plugins;
+    private static Collection<Class<? extends Plugin>> addMockTransportIfMissing(Settings settings,
+                                                                                 Collection<Class<? extends Plugin>> plugins) {
+        boolean settingExists = NetworkModule.TRANSPORT_TYPE_SETTING.exists(settings);
+        String transportType = NetworkModule.TRANSPORT_TYPE_SETTING.get(settings);
+        if (settingExists == false || MockTcpTransportPlugin.MOCK_TCP_TRANSPORT_NAME.equals(transportType)) {
+            if (plugins.contains(MockTcpTransportPlugin.class)) {
+                return plugins;
+            } else {
+                plugins = new ArrayList<>(plugins);
+                plugins.add(MockTcpTransportPlugin.class);
+                return plugins;
+            }
+        } else if (NioTransportPlugin.NIO_TRANSPORT_NAME.equals(transportType)) {
```

Comment:
I check in the `if` block. If is does not exist, we enter the `if` block and never go to the `else if` block.